packages/devtools_app/lib/src/debugger/file_search.dart

```diff
@@ -153,8 +153,41 @@ class FileQuery {
 
   bool get isEmpty => query.isEmpty;
 
+  bool get isMultiToken => query.contains(' ');
+
+  List<String> get tokens => query.split(' ');
+
+  bool isExactFileNameMatch(ScriptRef script) {
+    if (isEmpty) return false;
+
+    final fileName = _fileName(script.uri);
+
+    if (isMultiToken) {
+      return tokens.every((token) => fileName.caseInsensitiveContains(token));
+    }
+
+    return fileName.caseInsensitiveContains(query);
+  }
+
+  AutoCompleteMatch createExactFileNameAutoCompleteMatch(ScriptRef script) {
+    if (isEmpty) return AutoCompleteMatch(script.uri);
+
+    final fileName = _fileName(script.uri);
+    final fileNameIdx = script.uri.lastIndexOf(fileName);
```

Comment:
nit: rename to `fileNameIndex`. When I first read this, I thought it was `fileNameId` with a typo